pkg/splunk/enterprise/configuration.go

```diff
@@ -590,6 +590,13 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
 
 	// prepare defaults variable
 	splunkDefaults := "/mnt/splunk-secrets/default.yml"
+	// Check for apps defaults and add it to only the standalone or deployer/cm instances
```

Comment:
So if I have understood this right, to install apps on Standalones, CM & Deployer, setup `defaultsUrlApps`

- Am I right in saying that on LM, we still need to use `defaultsUrl`? For consistency purposes, should LM also be included?
- If both `defaultsUrlApps` & `defaultsUrl` are configured with `apps_locaton`, wonder how Ansible would treat this? I am guessing it would be much more efforts to expect operator to be smart enough to parse & detect this duplicate, & strip off one of the duplicates. Completely dropping `apps_location` from `defaultsUrl` would also need parsing logic.
- We should document the new argument & best practices in Examples.md

Reply:
> Am I right in saying that on LM, we still need to use defaultsUrl? For consistency purposes, should LM also be included?

So currently yes, but I can add LM to this list as well.  Or probably better, just exclude Indexer and SearchHead types since they will always be in the cluster.

> If both defaultsUrlApps & defaultsUrl are configured with apps_locaton, wonder how Ansible would treat this? I am guessing it would be much more efforts to expect operator to be smart enough to parse & detect this duplicate, & strip off one of the duplicates. Completely dropping apps_location from defaultsUrl would also need parsing logic.

I will test this and verifiy that ansible behaves correctly.

> We should document the new argument & best practices in Examples.md

We can add this to the documentation.  Though if we plan on making the Deployer its own CRD, we may not need this parameter.  So it maybe a temporary fix and might not need documenting if it's going away soon.  However we can pull the docs once that occurs.  I'll make the change.

Reply:
Validated in testing the use of multiple defaults.yml files works correctly, including with overlapping apps.